src/components/forms/fragments/exchangeApplication/ExchangeApplication.tsx

```diff
@@ -1,61 +1,102 @@
-              <option key={harbor.id} value={harbor.id}>
-                {harbor.name}
-              </option>
-            ))}
-        </Select>
-
-        <Row>
-          <Col sm={6}>
-            <Text
-              name={`pier`}
-              label={`page.berth.exchange_application.form.pier.title`}
-              placeholder={`page.berth.exchange_application.form.pier.placeholder`}
-            />
-          </Col>
-
-          <Col sm={6}>
-            <Text
-              name={`berthNumber`}
-              required
-              label={`page.berth.exchange_application.form.berth.title`}
-              placeholder={`page.berth.exchange_application.form.berth.placeholder`}
-            />
-          </Col>
-        </Row>
-      </div>
-    </div>
+    <Container className="vene-exchange-application">
+      <Row>
+        <Col>
+          <FormattedMessage id="page.berth.exchange_application.current_berth.title">
+            {txt => <h3 className="vene-exchange-application__heading">{txt}</h3>}
+          </FormattedMessage>
+          <FormattedMessage id="page.berth.exchange_application.current_berth.info_text">
+            {txt => <p className="vene-exchange-application__description">{txt}</p>}
+          </FormattedMessage>
+        </Col>
+      </Row>
+
+      <Row>
+        <Col>
+          <Select
+            name={`harborId`}
+            label="page.berth.exchange_application.form.current_harbour_area.label"
+            required
+          >
+            <option />
+            {harbors.size &&
+              harbors.map((harbor: HarborOption) => (
+                <option key={harbor.id} value={harbor.id}>
+                  {harbor.name}
+                </option>
+              ))}
+          </Select>
+        </Col>
+      </Row>
+
+      <Row>
+        <Col sm={6}>
+          <Text
+            name={`pier`}
+            label={`page.berth.exchange_application.form.pier.title`}
+            placeholder={`page.berth.exchange_application.form.pier.placeholder`}
+          />
+        </Col>
+
+        <Col sm={6}>
+          <Text
+            name={`berthNumber`}
+            required
+            label={`page.berth.exchange_application.form.berth.title`}
+            placeholder={`page.berth.exchange_application.form.berth.placeholder`}
+          />
+        </Col>
+      </Row>
+
+      <Row>
+        <Col>
+          <FormattedMessage id="page.berth.exchange_application.reason.title">
+            {txt => <h3 className="vene-exchange-application__heading">{txt}</h3>}
+          </FormattedMessage>
+          <FormattedMessage id="page.berth.exchange_application.reason.info_text">
+            {txt => <p className="vene-exchange-application__description">{txt}</p>}
+          </FormattedMessage>
+        </Col>
+      </Row>
+
+      <Row>
+        <Col>
+          <Select name={`reason`}>
+            <option />
+            {reasons &&
+              reasons.reduce<JSX.Element[]>(
```

Comment:
Are you using reduce here to filter out `null/undefined` values from reasons? I would try to minimize this kind of logic in the middle of template. Instead you could at the start of the function (before return) filter out the values you do not want and use the standard array-map here instead.